src/UpupUploader.tsx

```diff
@@ -13,18 +13,32 @@ interface UpupUploaderProps {
     setKey: (key: string) => void
     canUpload: boolean
     provider: Provider[],
+    compressible?: boolean,
     API_KEY?: string,
     APP_ID?: string,
     CLIENT_ID?: string,
 }
 
-
+/**
+ *
+ * @param client cloud provider client, ex: S3
+ * @param bucket bucket name
+ * @param setKey return the final name of the file, usually it has timestamp prefix
+ * @param canUpload to control when to upload the file , it has default false value
+ * @param provider whether the user want to upload files from internal storage or Google drive or both
+ * @param API_KEY you can get this from Google cloud console
+ * @param APP_ID the project ID inside Google cloud console
+ * @param CLIENT_ID the OAuth client ID
+ * @param compressible whether the user want to compress the file before uploading it or not. Default value is false
+ * @constructor
+ */
 export const UpupUploader: FC<UpupUploaderProps> = ({
                                                         client,
                                                         bucket,
                                                         setKey,
                                                         canUpload,
                                                         provider,
```

Comment:
Maybe pass the config object I mentioned in the other comment and the provider in the same prop

Reply:
@bassem97 will be tacking this in the next patch